src/main/groovy/edu/oregonstate/mist/inventory/resources/InventoryResource.groovy

```diff
@@ -69,17 +83,41 @@ class InventoryResource extends Resource {
     }
 
     /**
-     * Delete one inventory object by ID
+     * Create a new inventory object.
+     * @param newResultObject
+     * @return
+     */
+    @Timed
+    @POST
+    Response createInventory(@Valid ResultObject newResultObject) {
+        List<Error> errors = getErrors(newResultObject)
+
+        if (errors) {
+            Response.ResponseBuilder responseBuilder = Response.status(Response.Status.BAD_REQUEST)
+            return responseBuilder.entity(errors).build()
+        }
+        Inventory newInventory = resultObjectToInventory(newResultObject)
+        newInventory.id = newResultObject.data['id'] ?: randomUUID() as String
+
+        inventoryDAOWrapper.createInventory(newInventory)
+
+        ResultObject newCreatedInventory = new ResultObject(
```

Comment:
How about checking that inventoryDAOWrapper.getInventoryById(newInventory.id) is not null before building the ResultObject? You can then do some error handling.